src/hooks/searchParams.tsx

```diff
@@ -0,0 +1,31 @@
+import { useSearchParams } from 'react-router-dom';
+
+const enum SearchParams {
+  REDIRECTION = 'url',
+}
+
+/**
+ * Checks whether the redirection url search parameter is set, in which case the value should be passed to the back-end
+ * endpoints such that the final authentication (/auth) redirects to this URL (after e.g. password or e-mail)
+ *
+ * @returns { url } the redirection url to be passed with any authentication method
+ */
+export const useRedirection = () => {
+  const [searchParams] = useSearchParams();
+
+  let url: string | undefined;
+  try {
+    const urlParam = searchParams.get(SearchParams.REDIRECTION) ?? undefined;
```

Comment:
Is `?? undefined` necessary?

Reply:
Unfortunately yes, because the `URLSearchParams` API returns `null` if not found. So to avoid sending serialized `null`s it's converted to `undefined` using this.
See https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get#return_value